lib/src/main/java/com/auth0/jwt/JWTCreator.java

```diff
@@ -354,6 +354,32 @@ public Builder withClaim(String name, List<?> list) throws IllegalArgumentExcept
             return this;
         }
 
+        /**
+         * Add specific Claims to set as the Payload. If provided map is null then
+         * nothing is changed. If provided map contains a claim with null value then that
+         * claim will be removed from the payload
+         *
+         * @param payloadClaims the values to use as Claims in the token's payload.
+         * @throws IllegalArgumentException if payloadClaims is not well-formed and not having null values
+         * @return this same Builder instance.
+         */
+        public Builder withPayload(Map<?, ?> payloadClaims) throws IllegalArgumentException {
```

Comment:
Shouldn't this take a `Map<String, Object>`? As is, this does not compile (see build failure).